react-hot-button-component/src/index.jsx

```diff
@@ -0,0 +1,45 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+
+class HotButton extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = { counter: 0 };
+    this.handleClick = this.handleClick = this.handleClick.bind(this);
+  }
+
+  handleClick() {
+    this.setState(
+      { counter: this.state.counter + 1 }
+    );
+  }
+
+  render() {
+    const { counter } = this.state;
+    const bgcColors = ['purple', 'lpurple', 'red', 'orange', 'brown', 'yel', 'white'];
+    let index = 0;
+
+    if (counter < 3) {
+      index = 0;
+    } else if (counter < 6) {
+      index = 1;
+    } else if (counter < 9) {
+      index = 2;
+    } else if (counter < 12) {
+      index = 3;
+    } else if (counter < 15) {
+      index = 4;
+    } else if (counter < 18) {
+      index = 5;
+    } else if (counter >= 18) {
+      index = 6;
+    }
```

Comment:
Is there a pattern here? Maybe you could do some math to calculate the correct index instead of a bunch of `if` statements. 🤔